packages/calcite-components/src/components/block-section/block-section.scss

```diff
@@ -51,35 +47,58 @@
   &:hover {
     @apply text-color-1;
   }
-}
 
-.section-header .status-icon {
-  @apply self-end;
+  .label {
+    @apply flex items-center justify-center;
+  }
 }
 
 .section-header__text {
-  @apply mx-3
-    my-0
+  @apply my-0
     flex-auto;
 
   text-align: initial;
   word-wrap: anywhere;
 }
 
-.toggle--switch-container {
-  @apply flex items-center relative bg-transparent w-full;
+.toggle-container {
+  @apply flex items-center relative bg-transparent;
   word-break: break-word;
 
-  .focus-guard {
-    --calcite-label-margin-bottom: 0;
-    @apply absolute pointer-events-none;
-    inset-inline-end: 0;
-    margin-inline-start: theme("margin.1");
+  .toggle--switch__content {
+    @apply flex flex-auto items-center;
+  }
+
+  .icon--end {
+    @apply flex items-center;
+
+    color: var(--calcite-internal-block-section-icon-end-color, var(--calcite-color-text-3));
```

Comment:
Let's keep this if the internal CSS props help with #8594 integration. Otherwise, it might be cleaner to remove these here and add when the PR is synced w/ `main`.